enhancements/installer/infrastructure-apis-custom-trust.md

```diff
@@ -0,0 +1,236 @@
+
+On bootstrap host it reads the `<root>/tls/cloud-ca-cert.pem` for the custom trust, while in cluster it reads `ca-bundle.pem` key from ConfigMap specified in `.spec.cloudConfig`.
+
+Moving forward the MCO on bootstrap host will read the ConfigMap file `<root>/manifests/assets/manifests/infra-user-ca-bundle-config.yaml` for the trust bundle, and in cluster the operator will read the generated ConfigMap as described [above][#Infrastructure-API]. It should treat it similar to the Proxy trust bundle, i.e ensure it ends up in the list of machine trust [here][mco-additional-trust-list]
+
+### Test Plan
+
+One of the user stories is using this feature for testing AWS GovCloud using emulation. So that should provide appropriate CI for this feature.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+None
+
+### Upgrade / Downgrade Strategy
+
+This adds an optional input into already existing workflow and therefore there should be no effect to upgrade/downgrade strategy.
+
+### Version Skew Strategy
+
+This adds an optional input into already existing workflow and therefore there should be no effect to upgrade/downgrade strategy.
+
+## Implementation History
+
+### OpenStack workaround
+
+When OpenStack required custom trust bundle for user deployments, they added certificate key to the `.spec.cloudConfig` ConfigMap for example in Machine Config Operator [PR#1392][openstack-workaround-mco]. That workflow depends on a key in a config map that is meant to configure the `Kubernetes Core` bits interacting with cloud APIs like the kube-controller-manager. But that solution is not suitable for all our operators and also uses a less precise API. This new proposal is more widely implementable and with greater ease across platforms.
```

Comment:
will this approach take precedence over the existing api that openstack uses, if both are configured?

Reply:
this workaround and proposed changes can exist together and will not overrite each other as the workaround writes a different file on machine and configures the kubelet 's config to read that file.

Reply:
so what happens if both mechanisms are used/configured?  are both sets of CAs going to be loaded by the kubelet and the kube-controller-manager?